src/api/NodejsappBundlePart.ts

```diff
@@ -0,0 +1,203 @@
+
+/**
+ * Interface to represent the manifest data for a NODEJSAPP BundlePart.
+ *
+ * @export
+ * @interface INodejsappType
+ */
+interface INodejsappType {
+  nodejsapp: INodejsappContents;
+}
+
+/**
+ * Interface to represent the manifest data contents for a NODEJSAPP BundlePart.
+ *
+ * @export
+ * @interface INodejsappContents
+ */
+interface INodejsappContents {
+  name: string;
+  xmlns: string;
+  startscript: string;
+  profile: string;
+  lerunopts: string;
+}
+
+
+/**
+ * Class to represent a CICS NODEJSAPP BundlePart.
+ *
+ * @export
+ * @class NodejsappBundlePart
+ */
+export class NodejsappBundlePart extends BundlePart {
+
+  private static MAX_NODEJSAPP_LEN = 32;
+  private static MAX_PORT = 65535;
+  private nodejsappsDir: string;
+  private nodejsappFile: string;
+  private nodejsappProfile: string;
+  private nodejsappProfileLocal: string;
+  private partXML: INodejsappType;
+  private profile: string;
+
+  /**
+   * Constructor for creating a NodejsappBundlePart.
+   *
+   * @param {string} directory - The bundle directory.
+   * @param {string} name - The name of the NODEJSAPP BundlePart.
+   * @param {string} startscript - The path of the start script for the NODEJSAPP.
+   * @param {number} port - An optional port number to be added to the profile for the NODEJSAPP.
+   * @static
+   * @throws ImperativeError
+   * @memberof NodejsappBundlePart
+   */
+  constructor(directory: string, name: string, startscript: string, port: number) {
+    const partData = { name: "",
+                       type: "http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP",
+                       path: "" };
+
+    partData.name = name;
+    partData.path = "nodejsapps/" + name + ".nodejsapp";
+    super(directory, partData, false);
+
+    // Now validate the name
+    this.mangleName(NodejsappBundlePart.MAX_NODEJSAPP_LEN);
+    partData.name = this.getPartData().name;
+
+    // Now validate the port
+    this.validatePort(port);
+
+    this.nodejsappsDir = this.bundleDirectory + "/nodejsapps";
+    this.nodejsappFile = this.nodejsappsDir + "/" + partData.name + ".nodejsapp";
+    this.nodejsappProfile = this.nodejsappsDir + "/" + partData.name + ".profile";
+    this.nodejsappProfileLocal = "nodejsapps/" + partData.name + ".profile";
+
+    // Validate that the startscript resolves to something within the current directory
+    startscript = this.normalizeAndValidateFileReference(startscript);
+
+    this.createNodejsappXML(startscript);
+    this.createNodejsappProfile(port);
+  }
+
+  /**
+   * Get the NODEJSAPP BundlePart.
+   *
+   * @returns {INodejsappType}
+   * @throws ImperativeError
+   * @memberof NodejsappBundlePart
+   */
+  public getPart(): INodejsappType {
+    return this.partXML;
+  }
+
+  /**
+   * Get the NODEJSAPP BundlePart XML.
+   *
+   * @returns {string}
+   * @throws ImperativeError
+   * @memberof NodejsappBundlePart
+   */
+  public getPartXML(): string {
+    const parser = require("xml2json");
```

Comment:
I think this could be done as an import once at the top of the file rather than requiring it every time this method is called.